edx_reverification_block/xblock/reverification_block.py

```diff
@@ -0,0 +1,69 @@
+"""An XBlock for in-course reverification. """
+
+import pkg_resources
+from xblock.core import XBlock
+from xblock.fragment import Fragment
+
+
+def load(path):
+    """Load a resource from the package. """
+    data = pkg_resources.resource_string(__name__, path)
+    return data.decode("utf8")
+
+
+class ReverificationBlock(XBlock):
+    """An XBlock for in-course reverification. """
+
+    @property
+    def course_id(self):
+        return self._serialize_opaque_key(self.xmodule_runtime.course_id)  # pylint:disable=E1101
+
+    def student_view(self, context=None):
+        """Student view to render the re-verification link
+            This will render the url to display in lms.
+        """
+        # TODO: Need to change the hard coded url
+
+        anchor_str = "<a href='http://127.0.0.1:8003/verify_student/reverify/{course_id}/midterm/'>Reverify</a>"
+        return Fragment(unicode(anchor_str).format(self=self, course_id=self.get_course_id()))
+
+    def studio_view(self, context=None):
+        """Student view to render the re-verification link
+            This will render the url to display in lms.
+        """
+        # TODO: Need to change the hard coded url
+
+        anchor_str = "<a href='http://127.0.0.1:8003/verify_student/reverify/{course_id}/midterm/'>Reverify</a>"
+        return Fragment(unicode(anchor_str).format(self=self, course_id=self.get_course_id()))
+
+    @staticmethod
+    def workbench_scenarios():
+        return [
+            (
+                "Reverification Block",
+                load("static/xml/reverification_block_example.xml")
+            ),
+        ]
+
+    def _serialize_opaque_key(self, key):
+        """
+        Gracefully handle opaque keys, both before and after the transition.
+        https://github.com/edx/edx-platform/wiki/Opaque-Keys
+        Currently uses `to_deprecated_string()` to ensure that new keys
+        are backwards-compatible with keys we store in ORA2 database models.
+        Args:
+            key (unicode or OpaqueKey subclass): The key to serialize.
+        Returns:
+            unicode
+        """
+        if hasattr(key, 'to_deprecated_string'):
+            return key.to_deprecated_string()
+        else:
+            return unicode(key)
```

Comment:
Since `to_deprecated_string` has been deprecated, it's now safe to just use `unicode(key)`